pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -144,9 +144,16 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 					glog.V(3).Infof("Predicates failed for task <%s/%s> on node <%s>: %v",
 						task.Namespace, task.Name, node.Name, err)
 				} else {
-					workerLock.Lock()
-					predicateNodes = append(predicateNodes, node)
-					workerLock.Unlock()
+					clonedNode := node.Idle.Clone()
+					// Check for Resource Predicate
+					if task.InitResreq.LessEqual(clonedNode.Add(node.Releasing)) {
```

Comment:
let's check this before other predicates; if predicates failed, we can return it directly.